File: pastQuestions/fortinet02.py
grid = [[1, 1, 1], [1, 0, 1], [1, 1, 1]]

def findSafeWays(grid):
    n = len(grid) #3
    m = len(grid[0]) #3
    mod = (10**9) + 7

    dp = [[0] * m for _ in range(n)] 

    dp[0][0] = 1 if grid[0][0] == 1 else 0

    for i in range(n):
        for j in range(m):
            if grid[i][j] == 1:
                if i > 0:
                    dp[i][j] += dp[i-1][j]
                    # print("i",dp[i][j])
                if j > 0: 
                    dp[i][j] += dp[i][j-1]
                    # print("j",dp[i][j])
                dp[i][j]
                
    return dp[n-1][m-1] % mod

# Example usage
# findSafeWays(grid)
print(findSafeWays(grid))  # Output: 2

File: pastQuestions/fortinet02.js
function findSafeWays(grid) {
    const MOD = 1e9 + 7;
    const n = grid.length;
    const m = grid[0].length;
  
    // dp 배열 생성 및 초기화
    const dp = new Array(n).fill(0).map(() => new Array(m).fill(0));
  
    // 시작 지점 초기화
    dp[0][0] = grid[0][0] === 1 ? 1 : 0;
  
    // // i = 0 j = 1
    // for (let i = 0; i < n; i++) {
    //   for (let j = 0; j < m; j++) {
    //     if (grid[i][j] === 1) {
    //       if (i > 0) {
    //         dp[i][j] += dp[i - 1][j];
    //       }
    //       if (j > 0) {
    //         //dp[0][1] += dp[0][0]
    //         dp[i][j] += dp[i][j - 1];
    //       }
    //       dp[i][j] %= MOD;
    //     }
    //   }
    // }

    // return dp[n - 1][m - 1] % MOD;
  }
  
  // 예제 입력
  const grid = [
    [1, 1, 1],
    [0, 1, 1],
    [1, 1, 1]
  ];

  let k = 0

  console.log("==>",k %= 1e9 + 7)
  
  // 결과 출력
//   console.log(findSafeWays(grid)); // 출력: 2
  